src/lib/state.ts

```diff
@@ -24,46 +42,82 @@ interface CommandClassState {
   isSecure: boolean;
 }
 
-type EndpointState = Partial<Endpoint>;
-
-type DeviceClassState = Partial<
-  Modify<
-    DeviceClass,
-    {
-      basic: {
-        key: number;
-        label: string;
-      };
-      generic: {
-        key: number;
-        label: string;
-      };
-      specific: {
-        key: number;
-        label: string;
-      };
-    }
-  >
->;
+interface EndpointState {
+  nodeId: Endpoint["nodeId"];
+  index: Endpoint["index"];
+  installerIcon: Endpoint["installerIcon"];
+  userIcon: Endpoint["userIcon"];
+}
+
+interface DeviceClassState {
+  basic: {
+    key: number;
+    label: string;
+  };
+  generic: {
+    key: number;
+    label: string;
+  };
+  specific: {
+    key: number;
+    label: string;
+  };
+  mandatorySupportedCCs: DeviceClass["mandatorySupportedCCs"];
+  mandatoryControlledCCs: DeviceClass["mandatoryControlledCCs"];
+}
 
 interface ValueState extends TranslatedValueID {
   metadata: ValueMetadata;
   ccVersion: number;
   value?: any;
 }
 
-type NodeState = Partial<
-  Modify<
-    ZWaveNode,
-    {
-      deviceClass: DeviceClass | DeviceClassState;
-      commandClasses: CommandClassState[];
-      endpoints: EndpointState[];
-      values: ValueState[];
-    }
-  >
+interface NodeState0 {
+  nodeId: ZWaveNode["nodeId"];
+  index: ZWaveNode["index"];
+  installerIcon: ZWaveNode["installerIcon"];
+  userIcon: ZWaveNode["userIcon"];
+  status: ZWaveNode["status"];
+  ready: ZWaveNode["ready"];
+  isListening: ZWaveNode["isListening"];
+  isFrequentListening: ZWaveNode["isFrequentListening"];
+  isRouting: ZWaveNode["isRouting"];
+  maxBaudRate: ZWaveNode["maxBaudRate"];
+  isSecure: ZWaveNode["isSecure"];
+  version: ZWaveNode["version"];
+  isBeaming: ZWaveNode["isBeaming"];
+  manufacturerId: ZWaveNode["manufacturerId"];
+  productId: ZWaveNode["productId"];
+  productType: ZWaveNode["productType"];
+  firmwareVersion: ZWaveNode["firmwareVersion"];
+  zwavePlusVersion: ZWaveNode["zwavePlusVersion"];
+  nodeType: ZWaveNode["nodeType"];
+  roleType: ZWaveNode["roleType"];
+  name: ZWaveNode["name"];
+  location: ZWaveNode["location"];
+  deviceConfig: ZWaveNode["deviceConfig"];
+  label: ZWaveNode["label"];
+  neighbors: ZWaveNode["neighbors"];
+  endpointCountIsDynamic: ZWaveNode["endpointCountIsDynamic"];
+  endpointsHaveIdenticalCapabilities: ZWaveNode["endpointsHaveIdenticalCapabilities"];
+  individualEndpointCount: ZWaveNode["individualEndpointCount"];
+  aggregatedEndpointCount: ZWaveNode["aggregatedEndpointCount"];
+  interviewAttempts: ZWaveNode["interviewAttempts"];
+  interviewStage: ZWaveNode["interviewStage"];
+
+  deviceClass: DeviceClass | null;
```

Comment:
In API schema 0 this used to be `DeviceClass | undefined`, which caused the JSON serializer to remove the key. Now it is explicitly set to non-existing.